maestro-cli/src/main/java/maestro/cli/runner/TestSuiteInteractor.kt

```diff
@@ -0,0 +1,143 @@
+package maestro.cli.runner
+
+import maestro.Maestro
+import maestro.cli.device.Device
+import maestro.cli.model.FlowStatus
+import maestro.cli.util.PrintUtils
+import maestro.cli.view.TestSuiteStatusView
+import maestro.cli.view.TestSuiteStatusView.TestSuiteViewModel
+import maestro.orchestra.Orchestra
+import maestro.orchestra.yaml.YamlCommandReader
+import java.io.File
+
+object TestSuiteInteractor {
+
+    fun runTestSuite(
+        maestro: Maestro,
+        device: Device?,
+        input: File,
+        env: Map<String, String>,
+    ): SuiteResult {
+        return if (input.isFile) {
+            runTestSuite(
+                maestro, device, listOf(input), env
+            )
+        } else {
+            val flowFiles = input
+                .listFiles()
+                .filter {
+                    it.isFile
+                        && it.extension in setOf("yaml", "yml")
+                        && it.nameWithoutExtension != "config"
+                }
+                .toList()
+
+            runTestSuite(
+                maestro, device, flowFiles, env
+            )
+        }
+    }
+
+    fun runTestSuite(
+        maestro: Maestro,
+        device: Device?,
+        flows: List<File>,
+        env: Map<String, String>,
+    ): SuiteResult {
+        val flowResults = mutableListOf<SuiteResult.FlowResult>()
+
+        PrintUtils.message("Waiting for flows to complete...")
+        println()
+
+        var passed = true
+        flows.forEach { flowFile ->
+            val result = runFlow(flowFile, env, maestro)
+
+            if (result.status == FlowStatus.ERROR) {
+                passed = false
+            }
```

Comment:
This feels a bit fragile, though currently not an issue. The code assumes passed = true, then has a condition to set passed to false.
There is no way that the runFlow can except out so that passed is true but the test failed. Still it would be nice to tight this code down a bit more. For example by having a `successCount = 0` then adding 1 for each successful test. Then success will be `successCount == flows.count()`

Reply:
I find the counter to be quite less intuitive than a boolean as it simply has many more possible states and in fact is not going to be used as a counter